File: app/src/main/java/com/example/jy/jyweather/util/GpsUtil.java
package com.example.jy.jyweather.util;

import android.content.Context;
import android.location.LocationManager;

/**
 * Created by liyang
 * on 2019/1/4
 */
public class GpsUtil {

    /**
     * 判断定位是否打开
     */
    public static boolean isOpen(Context context) {
        LocationManager manager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
        boolean gps = manager.isProviderEnabled(LocationManager.GPS_PROVIDER);
        boolean network = manager.isProviderEnabled(LocationManager.NETWORK_PROVIDER);
        return gps || network;
    }
}

File: app/src/main/java/com/jy/weather/util/GpsUtil.kt
package com.jy.weather.util

import android.content.Context
import android.location.LocationManager

/**
 * Gps工具类
 *
 * Created by liyang
 * on 2019/1/4
 */
object GpsUtil {

    /**
     * 判断定位是否打开
     */
    fun isOpen(context: Context): Boolean {
        val manager = context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
        val gps = manager.isProviderEnabled(LocationManager.GPS_PROVIDER)
        val network = manager.isProviderEnabled(LocationManager.NETWORK_PROVIDER)
        return gps || network
    }
}
